third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -576,18 +575,22 @@ impl Interpreter {
             } => {
                 gas_meter.charge_native_function(cost, Option::<std::iter::Empty<&Value>>::None)?;
 
-                // Load the module that contains this function regardless of the traversal context.
-                //
-                // This is just a precautionary step to make sure that caching status of the VM will not alter execution
-                // result in case framework code forgot to use LoadFunction result to load the modules into cache
-                // and charge properly.
-                resolver
-                    .loader()
-                    .load_module(&module_name, data_store, module_store)
-                    .map_err(|_| {
-                        PartialVMError::new(StatusCode::FUNCTION_RESOLUTION_FAILURE)
-                            .with_message(format!("Module {} doesn't exist", module_name))
-                    })?;
+                // Note(George): when V2 loader fetches the function, the defining module is
+                // automatically loaded as well, and there is no need for preloading of a module
+                // into the cache like in V1 design.
+                if let Loader::V1(loader) = resolver.loader() {
+                    // Load the module that contains this function regardless of the traversal context.
```

Comment:
It this a hack due to traversal context? i.e. loader v1 makes sure all dependencies are loaded, and flush flushes everything, so why would smt not be available? 

if this is the reason, maybe we could call smt specific to the traversal context, not general load module (which I assume binds to filling the traversal context), but could be not worth it since this is all V1 anyway.